test/cypress/integration/my-services/update_service_name.js

```diff
@@ -0,0 +1,99 @@
+'use strict'
+
+const { getUserStubWithServiceName, getGatewayAccountsStub } = require('../../utils/common_stubs')
+const authenticatedUserId = 'authenticated-user-id'
+const serviceExternalId = 'service-external-id'
+const newServiceName = 'Updated Service'
+const serviceName = {
+  en: 'My Service'
+}
+const welshServiceName = {
+  en: 'My Service',
+  cy: 'Cymru'
+}
+
+function setupStubs (serviceName, stubs = []) {
+  cy.task('setupStubs', [
+    ...stubs,
+    getUserStubWithServiceName(authenticatedUserId, ['1'], serviceName, serviceExternalId),
+    getGatewayAccountsStub(1)
+  ])
+}
+
+function setUpStubWithWelshServiceName (welshServiceName) {
+  setupStubs(welshServiceName, [{
+    name: 'patchUpdateServiceNameSuccess',
+    opts: {
+      external_id: serviceExternalId,
+      serviceName: {
+        en: newServiceName,
+        cy: welshServiceName || ''
+      },
+      verifyCalledTimes: 1
+    }
+  }])
+}
+
+describe('Update service name', () => {
+  beforeEach(() => {
+    // keep the same session for entire describe block
+    Cypress.Cookies.preserveOnce('session')
+    Cypress.Cookies.preserveOnce('gateway_account')
+  })
+
+  describe('Edit a service name without a Welsh name', () => {
+    it('should display the my services page', () => {
+      cy.setEncryptedCookies(authenticatedUserId, 1)
+      setupStubs(serviceName)
+
+      cy.visit('/my-services')
+      cy.title().should('eq', 'Choose service - GOV.UK Pay')
+    })
+
+    it('should navigate to the edit service name form', () => {
+      setupStubs(serviceName)
+      cy.get('a').contains('Edit name').click()
+
+      cy.title().should('eq', 'Edit service name - GOV.UK Pay')
+      cy.get('#service-name').should('have.attr', 'value', 'My Service')
+      cy.get('#checkbox-service-name-cy').should('have.attr', 'aria-expanded', 'false')
+    })
+
+    it('should update service name to Updated Service', () => {
+      setUpStubWithWelshServiceName()
```

Comment:
I think this is ever so slightly confusing, as if we don't provide a `welshServiceName` it will setup the `getUserStubWithServiceName` to return an empty string for the `service_name` in the user response.

I think it would be better if this function just returned an object for the `patchUpdateServiceNameSuccess` which we then pass into the `setupStubs` call within the test.

Reply:
Done